### PR TITLE
build(webapp/deps): migrate to Angular v18

### PR DESCRIPTION
**Pull request by Char2sGu**

### CODE REVIEW

File: angular-client/src/index.html [hunk 1]
```diff
@@ -8,6 +8,6 @@
     <link href="https://api.tiles.mapbox.com/mapbox-gl-js/v1.4.1/mapbox-gl.css" rel="stylesheet" />
   </head>
   <body>
-    <app-context />
+    <app-context></app-context>
```
Comment (Char2sGu) [inline angular-client/src/index.html:11]: Custom Elements are not allowed to be self-closed. `<app-context />` was an invalid syntax, and after migration, this causes the build to fail

File: angular-client/src/app/app.module.ts [hunk 2]
```diff
@@ -190,19 +191,21 @@ import { BatteryInfoDisplay } from 'src/pages/charging-page/components/battery-i
...
-  bootstrap: [AppContext]
+  providers: [
+    DialogService,
+    MessageService,
+    provideHttpClient(withInterceptorsFromDi())
+  ]
 })
 export class AppModule {
-  constructor(
+  constructor (
```
Comment (Char2sGu) [inline angular-client/src/app/app.module.ts]: This formatting issue will be addressed later by forcing formatting and linting in Git Hooks

File: angular-client/src/app/app.module.ts [hunk 3]
```diff
@@ -190,19 +191,21 @@ import { BatteryInfoDisplay } from 'src/pages/charging-page/components/battery-i
...
     MatButtonModule,
     MatInputModule,
     ReactiveFormsModule
   ],
-  providers: [DialogService, MessageService],
-  bootstrap: [AppContext]
+  providers: [
+    DialogService,
+    MessageService,
+    provideHttpClient(withInterceptorsFromDi())
```
Comment (Char2sGu) [inline angular-client/src/app/app.module.ts]: Angular is going "standalone". The `HttpClientModule` is deprecated and replaced by the standalone provider factory `provideHttpClient`

Comment (Char2sGu) [reply angular-client/src/app/app.module.ts]: The new http interceptor api uses functional interceptors. In order to enable class-based interceptors, `withInterceptorsFromDi` is necessary.

**Review by RChandler234 — CHANGES_REQUESTED** [review]
This fucked with the sidebar design a bit somehow. I honestly like the new one better, but could you make it so the item highlight doesn't overflow on the right side
New:
image
Old:
image

**Comment by Char2sGu** [comment]
Sure. Since we bumped the version of Angular Material and PrimeNG, some updates in these packages might result in unexpected behaviors. I’ll look into that today.

**Comment by Char2sGu** [comment]
image

**Review by RChandler234 — APPROVED** [review]
LGTM 🔥

**Pull request merged**